src/main/fc/tasks.c

```diff
@@ -299,8 +299,133 @@ static void taskCameraControl(uint32_t currentTime)
 }
 #endif
 
+#define DEFINE_TASK(taskNameParam, subTaskNameParam, checkFuncParam, taskFuncParam, desiredPeriodParam, staticPriorityParam) {  \
+    .taskName = taskNameParam, \
+    .subTaskName = subTaskNameParam, \
+    .checkFunc = checkFuncParam, \
+    .taskFunc = taskFuncParam, \
+    .desiredPeriodUs = desiredPeriodParam, \
+    .staticPriority = staticPriorityParam \
+}
+
+// Task info in .bss (unitialised data)
+task_t tasks[TASK_COUNT];
+
+// Task ID data in .data (initialised data)
+task_id_t task_ids[TASK_COUNT] = {
```

Comment:
Another possibility is to use `const task_def[]` and copy it into `tasks[]` on init

Reply:
Many ways to skin a cat. I avoided the copy at the expense of one pointer per task.